lib/sqlalchemy/dialects/postgresql/array.py

```diff
@@ -127,15 +149,21 @@ def __init__(self, clauses, **kw):
                     if main_type.dimensions is not None
                     else 2
                 ),
-            )
+            )  # type: ignore[assignment]
```

Comment:
strange that we have this type error. self.type is a typeengine

Reply:
Not sure, but it might be due to `postgresql.array` being generic (`class array(expression.ExpressionClauseList[_T]):`) whereas the base `ARRAY` is not (`class ARRAY(SchemaEventTarget, Indexable, Concatenable, TypeEngine[Sequence[Any]]):`. So when assigning an `ARRAY()` value to `self.type`(which is a `TypeEngine[_T]`, the generic is lost, isn't it?